clients/src/main/java/org/apache/kafka/common/config/SaslConfigs.java

```diff
@@ -32,6 +32,11 @@
         + "Only GSSAPI is enabled by default.";
     public static final List<String> DEFAULT_SASL_ENABLED_MECHANISMS = Collections.singletonList(GSSAPI_MECHANISM);
 
+    /** Jaas configuration file format is described <a href="http://docs.oracle.com/javase/7/docs/technotes/guides/security/jgss/tutorials/LoginConfigFile.html">here</a> */
+    public static final String SASL_JAAS_CONFIG = "sasl.jaas.config";
```

Comment:
@harshach Do you mean add new properties for passing in keytab and principal name? That would be a very mechanism-specific approach. We would then require other parameters for other mechanisms (eg. username/password for PLAIN) And since mechanisms can be dynamically plugged in without changing Kafka, it is hard to define generic properties. As described in https://cwiki.apache.org/confluence/display/KAFKA/KIP-85%3A+Dynamic+JAAS+configuration+for+Kafka+clients, one of the other motivations of this approach is to enable Kafka clients to be run without a physical file containing jaas configuration.

If you defined new properties for keytab and principal, how would they make it into a login configuration? Don't you still need to create a JAAS Configuration in order to use a Kerberos login module? The property names are dependent on the login module (IBM JRE uses different option name for keytab from Oracle). It feels more flexible and future-proof if the login configuration is simply passed in as a property and Kafka doesn't need to understand options used by different login modules.